### DISCUSSION TITLE
[V5] how to get values from ManyToMany relationship on the Middleware?

### DISCUSSION THREAD

**Discussion by vicainelli** [adonisjs/core#2190] [Help] [answered]
Hey all, I'm writing a test for an API and a facing this problem when I trying to put a middleware to check if the User can access the Establishment

```ts
// app/Models/User.ts
@manyToMany(() => Establishment)
public establishments: ManyToMany<typeof Establishment>
```

But I got `undefined` when I try to fetch the user establishments on the middleware: 
```ts
// app/Middleware/EstablishmentUser.ts


public async handle({ request, response, auth }: HttpContextContract, next: () => Promise<void>) {
    const userEstablishments = await auth.user?.establishments
   
   // do something to check if user cannot access
    if (check) return response.status(401)
    
    await next()
  }


```

And works well when I fetch from the endpoint using `.preload('establishments')`

**Comment by McSneaky** [comment, marked as answer]
```js
// await auth.user.preload('establishments') (deprecated)
await auth.user.load('establishments')
const userEstablishments = auth.user.establishments
```
It works if you preload it first in the middleware

**Reply by vicainelli** [reply]
Worked, thanks @McSneaky

**Reply by deiby-dayans** [reply]
I don't know the reason, but the preload method does not work for me.
In my case, I have the manyTomany in both sides (User, Role) for some reason the preload does not work in the User side.
I did it by using the .load() method while getting the specific user.
